lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -1005,27 +998,26 @@ def _AssignCpuAffinity(cls, cpu_mask, process_id, thread_dict):
       for i, vcpu in enumerate(cpu_list):
         cls._SetProcessAffinity(thread_dict[i], vcpu)
 
-  def _GetVcpuThreadIds(self, instance_name):
+  @_with_qmp
+  def _GetVcpuThreadIds(self, instance):
     """Get a mapping of vCPU no. to thread IDs for the instance
 
     @type instance_name: string
-    @param instance_name: instance in question
+    @param instance: instance in question
     @rtype: dictionary of int:int
     @return: a dictionary mapping vCPU numbers to thread IDs
 
     """
     result = {}
-    output = self._CallMonitorCommand(instance_name, self._CPU_INFO_CMD)
-    for line in output.stdout.splitlines():
-      match = self._CPU_INFO_RE.search(line)
-      if not match:
-        continue
-      grp = [int(g) for g in match.groups()]
-      result[grp[0]] = grp[1]
+    cpu_info = self.qmp.GetFastCpuInformation()
+
+    result = {}
```

Comment:
is `result` here initialized twice?

Reply:
Yes, you are right. Good catch! Must have been late :-)